### PR TITLE
Add endpoint to show source of resolved resources

### LINKED ISSUE #4462: Endpoint to return the source of a resource through a resolver
When resolving a resource through one or many resolvers in a request such as:
```
/v1/resolvers/org/proj/_/resource-id
```
I would like to be able to receive the original payload submitted (the source), in the same manner that I can when looking up a resource normally

### PR DESCRIPTION
**Pull request by shinyhappydan**
Fixes #4462 

image

### CODE REVIEW

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala
Comment (shinyhappydan) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala:1]: All the changes here are just renames

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResolversRoutes.scala [hunk 1]
```diff
@@ -167,23 +176,30 @@ final class ResolversRoutes(
...
       parameter("showReport".as[Boolean].withDefault(default = false)) { showReport =>
-        def emitResult[R: JsonLdEncoder](io: IO[MultiResolutionResult[R]]) =
+        def emitResult[R: JsonLdEncoder](io: IO[MultiResolutionResult[R]]) = {
           if (showReport)
             emit(io.map(_.report).attemptNarrow[ResolverRejection])
+          else if (asSource)
+            emit(io.map(_.value.source).attemptNarrow[ResolverRejection])
           else
             emit(io.map(_.value.jsonLdValue).attemptNarrow[ResolverRejection])
+        }
```
Comment (shinyhappydan) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResolversRoutes.scala:196]: There is a question here about combinations of `showReport` and `asSource` but I think this works and personally I don't think the refactor needed is worth it

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResolversRoutes.scala [hunk 2]
```diff
@@ -167,23 +176,30 @@ final class ResolversRoutes(
       }
     }
 
-  private def resolve(resourceSegment: IdSegmentRef, projectRef: ProjectRef, resolverId: Option[IdSegment])(implicit
+  private def resolve(
+      resourceSegment: IdSegmentRef,
+      projectRef: ProjectRef,
+      resolverId: Option[IdSegment],
+      asSource: Boolean = false
```
Comment (dantb) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResolversRoutes.scala:183]: meganit - could be mandatory

**Review by dantb — APPROVED** [review]

**Comment by imsdu** [comment]
The endpoint need to be documented otherwise it looks good to me

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResolversRoutes.scala [hunk 3]
```diff
@@ -167,23 +186,48 @@ final class ResolversRoutes(
...
-            emit(io.map(_.value.jsonLdValue).attemptNarrow[ResolverRejection])
-
-        resolverId.fold(emitResult(multiResolution(resourceSegment, projectRef))) { resolverId =>
-          emitResult(multiResolution(resourceSegment, projectRef, resolverId))
+    authorizeFor(project, Permissions.resources.read).apply {
+      def emitResult[R: JsonLdEncoder](io: IO[MultiResolutionResult[R]]) = {
+        output match {
+          case ResolvedResourceOutputType.Report => emit(io.map(_.report).attemptNarrow[ResolverRejection])
+          case ResolvedResourceOutputType.JsonLd => emit(io.map(_.value.source).attemptNarrow[ResolverRejection])
+          case ResolvedResourceOutputType.Source => emit(io.map(_.value.jsonLdValue).attemptNarrow[ResolverRejection])
```
Comment (dantb) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResolversRoutes.scala:202]: Much clearer 🙌  but I think the last two are the wrong way round

**Review by dantb — CHANGES_REQUESTED** [review]

**Review by dantb — APPROVED** [review]

**Review by dantb — APPROVED** [review]

**Pull request merged**